### PR TITLE
Add integration tests for retrieving dir info using file name

### PR DESCRIPTION
**Pull request by medb**
Add test that retrieves non-empty regular and implicit directory info using its object name (i.e. without trailing `/`)

### CODE REVIEW

File: gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemNewIntegrationTest.java [hunk 1]
```diff
@@ -304,7 +331,35 @@ public void getDirInfo_sequential() throws Exception {
   }
 
   @Test
-  public void getDirInfo_parallel() throws Exception {
+  public void getDirInfo_objectName_parallel() throws Exception {
```
Comment (mprashanthsagar) [inline gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemNewIntegrationTest.java:334]: nit: do we want to re-order tests ?

**Review by mprashanthsagar — APPROVED** [review]

File: gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemNewIntegrationTest.java (hunk 1 above)
Comment (medb) [reply gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemNewIntegrationTest.java:386]: I order them logically - from "simpler" `../name` test to more complex `../name/` test, but I can reorder them alphabetically.

**Review by mprashanthsagar — APPROVED** [review]
